### ISSUE TITLE
2.3.3 Generating does not working in the Eastern hemisphere

### ISSUE THREAD

**Issue by dKubasa** [jakubroztocil/rrule#247]
after update  on server 2.2.0 -> 2.3.3  rule generate only 1 date.

```
new RRule({
          freq: RRule.WEEKLY,
          interval: 1,
          dtstart: new Date(startEvent),
          until: new Date(endSearch)
        }) 
```

server : Ubuntu 16.04.4 LTS

But locally i have windows 10 and  2.3.3.  work fine

**Comment by arolson101** [comment]
Could you post the specific values for startEvent, endSearch, and what the output is on the server?  Also, what timezone is the server set to?

**Comment by davidgoli** [comment]
@dKubasa we'd love to take a look at this, but please provide the specifics @arolson101 requested. Thank you!

**Comment by dKubasa** [comment]
@davidgoli @arolson101 

Wrong information. Widows 10 locally also does not work.
This test is on windows 10.

2.3.3
TimeZone:  Central European Standard Time

```
startEvent = 1533895200000;
endSearch = 1543618799999;

new RRule({
          freq: RRule.WEEKLY,
          interval: 1,
          dtstart: new Date(startEvent),
          until: new Date(endSearch)
        }) 
```

RESULT:
`[ 2018-08-10T10:00:00.000Z ]`

2.2.0
TimeZone:  Central European Standard Time
```
startEvent = 1533895200000;
endSearch = 1543618799999;

new RRule({
          freq: RRule.WEEKLY,
          interval: 1,
          dtstart: new Date(startEvent),
          until: new Date(endSearch)
        }) 
```

RESULT:
```
[ 2018-08-10T10:00:00.000Z,
  2018-08-17T10:00:00.000Z,
  2018-08-24T10:00:00.000Z,
  2018-08-31T10:00:00.000Z,
  2018-09-07T10:00:00.000Z,
  2018-09-14T10:00:00.000Z,
  2018-09-21T10:00:00.000Z,
  2018-09-28T10:00:00.000Z,
  2018-10-05T10:00:00.000Z,
  2018-10-12T10:00:00.000Z,
  2018-10-19T10:00:00.000Z,
  2018-10-26T10:00:00.000Z,
  2018-11-02T11:00:00.000Z,
  2018-11-09T11:00:00.000Z,
  2018-11-16T11:00:00.000Z,
  2018-11-23T11:00:00.000Z,
  2018-11-30T11:00:00.000Z ]
```

**Comment by davidgoli** [comment]
I can confirm that I see this when I set my machine's timezone to Central European Standard Time. In fact, all timezones in the Western hemisphere seem to work correctly, but timezones in the Easter hemisphere do not.

Also note the times generated by 2.2.0 show the "one hour off" bug that has been fixed. But that does you no good if you don't get the correct dates!

**Comment by davidgoli** [comment]
In fact, a large number of tests currently fail using Eastern hemisphere time zones. I will set up the CI server to perform test runs under multiple time zones to catch this sort of issue in the future.

**Comment by davidgoli** [comment]
@dKubasa This should be fixed with v2.3.4. Let me know if you continue to experience issues!
